js/node_shared/XMLHttpRequest.js

```diff
@@ -13,6 +13,26 @@
  * @author Dan DeFelippi <dan@driverdan.com>
  * @contributor David Ellis <d.f.ellis@ieee.org>
  * @license MIT
```

Comment:
Is this whole file copied, or is it modified a fair amount? I would recommend having a note so we know what's what (like precisely what is just inbound versus derived versus ours and unrelated -- best of the last isn't in the same file but might make sense)